src/models/booking.cr

```diff
@@ -125,11 +124,15 @@ class Booking
 
   def current_history : Array(History)
     state = current_state
-    history_column.value([] of History).dup.tap do |booking_history|
+    new_history = history_column.value([] of History).dup.tap do |booking_history|
       if booking_history.empty? || booking_history.last.state != state
         booking_history << History.new(state, Time.local.to_unix, @utm_source) unless state.unknown?
       end
     end
+
+    Log.error { "History contains more than 3 events" } if new_history.size > 3
```

Comment:
Could you please add the booking id to this log

Reply:
You can also do this check inside of the block above, saving the assignment to `new_history`